### DISCUSSION TITLE
Photoprism can't see my GPU even though it's passed through?

### DISCUSSION THREAD

**Discussion by doctorttt** [photoprism/photoprism#3506] [General]
I researched this topic extensively on the web and followed the instructions I found, but I still can't get my GPU to show up in Photoprism. See below the relevant settings. I'm not sure what else needs to be done. 

- I'm using Unraid and ```photoprism/photoprism``` latest docker image
- I added ```--runtime=nvidia``` to the docker container.
- Added ```NVIDIA_DRIVER_CAPABILITIES``` variable with ```all```
- Added ```NVIDIA_VISIBLE_DEVICES``` variable with my gpu's GUID. I also tried ```all```
- Added ```PHOTOPRISM_FFMPEG_BUFFERS``` variable with ```64```
- Added ```PHOTOPRISM_FFMPEG_ENCODER``` variable with ```h264_nvenc```. I even tried ```nvidia```.
- Added ```PHOTOPRISM_INIT``` variable with ```tensorflow```
- Added ```PHOTOPRISM_FFMPEG_BITRATE``` variable with ```50```
- ```nvidia-smi``` sees no related process

**Comment by lastzero** [comment]
Could you share additional information, such as what software versions you use and if other software can use the GPU? Maybe additional drivers must be installed on the host?

**Comment by doctorttt** [comment]
> Could you share additional information, such as what software versions you use and if other software can use the GPU? Maybe additional drivers must be installed on the host?

I'm using Nvidia driver 535.54.03 on my host, and I'm 100% sure there is no issue with my GPU because I can pass it to Jellyfin, Plex, and Frigate without any issues.

**Reply by lastzero** [reply]
Is PhotoPrism running under the same user ID? Could be a permissions issue otherwise.

**Reply by doctorttt** [reply]
How is this a permission issue when I can pass the same gpu to Jellyfin Plex, and Frigate? I even gave Photoprism root / privilege access to my host and still doesn't work. 

I'm using Build 230625-17242fb07

**Reply by lastzero** [reply]
Could it be that these services are running with different permissions or under a different user/group ID?

In case you haven't seen it yet, here is our Nvidia config example that we successfully tested with Ubuntu Linux as host:

https://dl.photoprism.app/docker/nvidia/docker-compose.yml

I'm not entirely sure how the Nvidia Toolbox integration works, unfortunately. As far as I know, the guest container is automatically configured and additional drivers and tools may be installed there if needed. Since we are using a relatively new Ubuntu base image (23.04 Lunar Lobster), while other Docker images are often based on older versions or Debian, it could be that the installed version of the NVIDIA Container Toolkit just doesn't support it yet. But that's just a wild guess.

Also note that you can currently only use GPUs for video transcoding and RAW image conversion. Using it with TensorFlow is also possible in principle, but requires custom-built TensorFlow binaries with support for your GPU, which we currently cannot provide.

This is mainly because we didn't have an Nvidia GPU to test with when we built the binaries (which took about a week, so is very time-consuming and requires experience with C/C++ compilers). Once we have updated to the latest TensorFlow version, we will also provide binaries with Nvidia support.

**Reply by doctorttt** [reply]
Ok, thanks for the info. One question - when using ```PHOTOPRISM_INIT: tensorflow``` does this use GPU or CPU? I remember seeing ```gpu tensorflow``` and ```intel tensorflow```. 

https://docs.photoprism.app/getting-started/advanced/transcoding/

**Reply by lastzero** [reply]
This installs a TensorFlow version with AVX or AVX2 support, if your CPU supoorts it. It could also install a version with GPU support, but as I said, we don't have one (just for the Nvidia Jetson, see https://dl.photoprism.app:/tensorflow/).
